fish-rust/src/common.rs

```diff
@@ -1742,44 +1763,24 @@ pub type Cleanup<T, F, C> = ScopeGuard<T, F, C>;
 ///
 /// // hello will be written first, then goodbye.
 /// ```
-pub struct ScopeGuard<T, F: FnOnce(&mut T), C> {
+pub struct ScopeGuard<T, F: FnOnce(&mut T)> {
```

Comment:
FYI I don't object to you merging this since I recognize implementing `scoped_cleanup` is hard in Rust. But it ends up looking pretty big and confusing in Rust Analyzer, (`&mut &mut &mut...`) so I will likely take a run at simplifying this.

<img width="674" alt="Screenshot 2023-05-21 at 2 05 48 PM" src="https://github.com/fish-shell/fish-shell/assets/920838/0040fe87-9711-429a-b561-e2d50167294a">